share/spack/gitlab/cloud_pipelines/configs/cray-sles/ci.yaml

```diff
@@ -0,0 +1,295 @@
+ci:
+  pipeline-gen:
+  - build-job-remove:
+      image: no-image
+  - build-job:
+      script+:
+      # AWS runners mount E4S public key (verification), UO runners mount public/private (signing/verification)
+      - if [[ -r /etc/protected-runner/e4s.gpg ]]; then spack gpg trust /etc/protected-runner/e4s.gpg; fi
+      # UO runners mount intermediate ci public key (verification), AWS runners mount public/private (signing/verification)
+      - if [[ -r /etc/protected-runner/intermediate_ci_signing_key.gpg ]]; then spack gpg trust /etc/protected-runner/intermediate_ci_signing_key.gpg; fi
+      - if [[ -r /etc/protected-runner/spack_public_key.gpg ]]; then spack gpg trust /etc/protected-runner/spack_public_key.gpg; fi
+  - match_behavior: first
```

Comment:
I am not super excited about all of this being duplicated between the `linux/ci.yaml` configs and the `cray-sles/ci.yaml` configs and the `cray/ci.yaml` configs.

I think this file can be pretty easily merged with the top level `linux/ci.yaml` and the `cray-sles/zen4` could be moved to `linux/cray-sles-zen4/ci.yaml` and include

```yaml
ci:
  pipeline-gen:
  - build-job-remove:
      image: no-image-for-cray
  - build-job:
      tags: ["pe-gnu-amd", "cray-sles-zen4"]
```

Reply:
How do we move the following to linux/ci.yaml? `/etc/protected-runner/` is a path that we have to use specifically on the Cray machines because we cannot use the path that is used everywhere else `/mnt/key`. The paths are different.

```
  pipeline-gen:
  - build-job-remove:
      image: no-image
  - build-job:
      script+:
      # AWS runners mount E4S public key (verification), UO runners mount public/private (signing/verification)
      - if [[ -r /etc/protected-runner/e4s.gpg ]]; then spack gpg trust /etc/protected-runner/e4s.gpg; fi
      # UO runners mount intermediate ci public key (verification), AWS runners mount public/private (signing/verification)
      - if [[ -r /etc/protected-runner/intermediate_ci_signing_key.gpg ]]; then spack gpg trust /etc/protected-runner/intermediate_ci_signing_key.gpg; fi
      - if [[ -r /etc/protected-runner/spack_public_key.gpg ]]; then spack gpg trust /etc/protected-runner/spack_public_key.gpg; fi
```

Reply:
Please see the latest update where I parameterize the location of GPG keys via `CI_GPG_KEY_ROOT` env variable. For `e4s-cray-rhel` and `e4s-cray-sles` this gets set to `/etc/protected-runner` via `configs/cray-{rhel,sles}/ci.yml`